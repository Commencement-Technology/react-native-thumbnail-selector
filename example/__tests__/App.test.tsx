/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

// Note: import explicitly to use the types shipped with jest.
import {it} from '@jest/globals';

it('renders correctly', () => {
  const component = renderer.create(<App />);
  expect(component.toJSON()).toMatchSnapshot();
});
