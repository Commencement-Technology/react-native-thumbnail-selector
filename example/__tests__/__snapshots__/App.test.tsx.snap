// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#001C34",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      {
        "marginHorizontal": 8,
      }
    }
  >
    <Image
      source={
        {
          "uri": "https://reactnative.dev/img/tiny_logo.png",
        }
      }
      style={
        {
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "height": "50%",
          "width": "100%",
        }
      }
    />
    <RCTScrollView
      ListHeaderComponent={[Function]}
      data={
        [
          {
            "key": "index",
            "value": "0",
          },
          {
            "key": "caption",
            "value": "David",
          },
          {
            "key": "imageSrc",
            "value": "{"uri":"https://reactnative.dev/img/tiny_logo.png"}",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      initialNumToRender={3}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEnabled={false}
      scrollEventThrottle={0.0001}
      stickyHeaderIndices={[]}
      style={
        {
          "backgroundColor": "#2D2C52",
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
        }
      }
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          collapsable={false}
          onLayout={[Function]}
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "padding": 8,
              }
            }
          >
            <Text
              style={
                {
                  "color": "white",
                  "fontWeight": "700",
                  "marginRight": 4,
                }
              }
            >
              OFF
            </Text>
            <RCTSwitch
              accessibilityRole="switch"
              onChange={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "height": 31,
                  "width": 51,
                }
              }
              value={false}
            />
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              {
                "flexDirection": "column",
                "padding": 4,
              }
            }
          >
            <Text
              style={
                {
                  "color": "white",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              index
            </Text>
            <Text
              style={
                {
                  "color": "white",
                  "fontSize": 14,
                }
              }
            >
              0
            </Text>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              {
                "flexDirection": "column",
                "padding": 4,
              }
            }
          >
            <Text
              style={
                {
                  "color": "white",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              caption
            </Text>
            <Text
              style={
                {
                  "color": "white",
                  "fontSize": 14,
                }
              }
            >
              David
            </Text>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              {
                "flexDirection": "column",
                "padding": 4,
              }
            }
          >
            <Text
              style={
                {
                  "color": "white",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              imageSrc
            </Text>
            <Text
              style={
                {
                  "color": "white",
                  "fontSize": 14,
                }
              }
            >
              {"uri":"https://reactnative.dev/img/tiny_logo.png"}
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <react-native-thumbnail-selector
    initialIndex={0}
    onSelect={[Function]}
    thumbnails={
      [
        {
          "caption": "David",
          "imageSrc": {
            "uri": "https://reactnative.dev/img/tiny_logo.png",
          },
        },
        {
          "caption": "Brian",
          "imageSrc": {
            "uri": "https://randomuser.me/api/portraits/women/67.jpg",
          },
        },
        {
          "caption": "Gene",
          "imageSrc": {
            "uri": "https://reactnative.dev/img/tiny_logo.png",
          },
        },
        {
          "caption": "Jose",
          "imageSrc": {
            "uri": "https://reactnative.dev/img/tiny_logo.png",
          },
        },
        {
          "caption": "Jon",
          "imageSrc": {
            "uri": "https://reactnative.dev/img/tiny_logo.png",
          },
        },
        {
          "caption": "Craig",
          "imageSrc": {
            "uri": "https://reactnative.dev/img/tiny_logo.png",
          },
        },
        {
          "caption": "Sean",
          "imageSrc": {
            "uri": "https://reactnative.dev/img/tiny_logo.png",
          },
        },
        {
          "caption": "Anim cillum voluptate mollit esse reprehenderit anim enim elit veniam duis cupidatat irure.",
          "imageSrc": {
            "uri": "https://reactnative.dev/img/tiny_logo.png",
          },
        },
        {
          "caption": "Enim incididunt",
          "imageSrc": {
            "uri": "https://reactnative.dev/img/tiny_logo.png",
          },
        },
        {
          "caption": undefined,
          "imageSrc": {
            "uri": "https://reactnative.dev/img/tiny_logo.png",
          },
        },
        {
          "caption": "",
          "imageSrc": {
            "uri": "https://reactnative.dev/img/tiny_logo.png",
          },
        },
        {
          "caption": "Cleveland Guardians",
          "imageSrc": {
            "testUri": "../../../assets/cg.png",
          },
        },
        {
          "caption": "Milwaukee Brewers",
          "imageSrc": {
            "testUri": "../../../assets/mb.png",
          },
        },
        {
          "caption": "New York Mets",
          "imageSrc": {
            "testUri": "../../../assets/nym.png",
          },
        },
        {
          "caption": "New York Yankees",
          "imageSrc": {
            "testUri": "../../../assets/nyy.png",
          },
        },
        {
          "caption": "Oakland A's",
          "imageSrc": {
            "testUri": "../../../assets/oa.png",
          },
        },
        {
          "caption": "San Diego Padres",
          "imageSrc": {
            "testUri": "../../../assets/sdp.png",
          },
        },
      ]
    }
    toggle={[Function]}
  />
</RCTSafeAreaView>
`;
